tests/integration/components/bs-modal-test.js

```diff
@@ -246,4 +246,65 @@ module('Integration | Component | bs-modal', function (hooks) {
 
     assert.dom('.modal').isVisible('Modal is visible again');
   });
+
+  testNotBS3('it animates opening the modal', async function (assert) {
+    this.set('open', false);
+    await render(hbs`<BsModal @open={{this.open}}>Hello world!</BsModal>`);
+
+    this.set('open', true);
+    await waitFor('.modal.show');
+    assert.dom('.modal').hasClass('show');
+    assert.dom('.modal').hasStyle({ display: 'block' });
+
+    await waitUntil(() => {
+      return find('.modal').getAnimations().length > 0;
+    });
+    assert.ok(
+      find('.modal')
+        .getAnimations()
+        .find((animation) => animation.transitionProperty === 'opacity'),
+      'modal opening is animated'
+    );
+  });
+
+  testNotBS3('it animates closing the modal', async function (assert) {
+    await render(hbs`
+      <BsModal as |modal|>
+        <button {{on "click" modal.close}} type="button">close</button>
+      </BsModal>
+    `);
+
+    // wait until modal is shown and opening transition is finished
+    await waitFor('.modal.show');
+    await Promise.all(
+      find('.modal')
+        .getAnimations()
+        .map((animation) => animation.finished)
+    );
+
+    // keep reference to modal element to compare with event target after modal has been destroyed
```

Comment:
Seems `modalEl` is not used "after modal has been destroyed", so this comment looks a bit misleading, or do I misunderstand?